File: Python/933. Number of Recent Calls.py
class RecentCounter(object):

    def __init__(self):
        self.arr = []
        self.base = 0

    def ping(self, t):
        """
        :type t: int
        :rtype: int
        """
        self.arr.append(t)
        if t - self.arr[self.base] <= 3000:
            return len(self.arr) - self.base
        else:
            while self.base < len(self.arr) and t - self.arr[self.base] > 3000:
                self.base += 1
            return len(self.arr) - self.base



# Your RecentCounter object will be instantiated and called as such:
# obj = RecentCounter()
# param_1 = obj.ping(t)

File: Java/933. Number of Recent Calls.java
package Java;

import java.util.ArrayList;
import java.util.List;

class RecentCounter {
    private int base;
    private List<Integer> arr;
    public RecentCounter() {
        arr = new ArrayList<Integer>();
        base = 0;
    }
    
    public int ping(int t) {
        arr.add(t);
        if (t - arr.get(base) <= 3000) return arr.size() - base;
        else{
            while (base < arr.size() && t - arr.get(base) > 3000) base++;
            return arr.size() - base;
        }
    }
}

/**
 * Your RecentCounter object will be instantiated and called as such:
 * RecentCounter obj = new RecentCounter();
 * int param_1 = obj.ping(t);
 */